src/main.rs

```diff
@@ -1,13 +1,13 @@
 mod proxy;
 
-use config::Config;
+// use config::Config;
 use hyper::server::conn::AddrStream;
 use serde::Deserialize;
-use std::{convert::Infallible, net::SocketAddr};
+use std::{convert::Infallible, net::SocketAddr, sync::Arc};
 
 /// Configuration of Comproxity.
 #[derive(Deserialize)]
-struct ComproxityConfig {
+pub struct Config {
```

Comment:
Since it's now shared to other modules and may grow in the future, move it to a separate module and make `pub(crate)`

Reply:
It may also be worth adding a type-def in the module for `Arc<Config>`